lib/s3storage.php

```diff
@@ -133,7 +133,12 @@ public function writeObject($urn, $stream) {
 		try {
 			$uploader->upload();
 		} catch (AwsException $e) {
-			throw new ObjectStoreWriteException($e->getMessage(), $e->getCode(), $e);
+			/**
+			 * If the error is from AwsException then just wrap the aws error message
+			 * to our exception.
+			 */
+			\OC::$server->getLogger()->logException($e);
+			throw new ObjectStoreWriteException($e->getAwsErrorMessage(), $e->getCode(), $e);
```

Comment:
I'm not sure what the `$e->getCode()` will return. We could use the `$e->getStatusCode()` instead, so we could distinguish some errors if needed.

Reply:
Done. Replaced the `getCode` with `getStatusCode`. Yes you are right, its better to rely on the status code.